partcad/src/partcad/part_factory_enrich.py

```
@@ -13,76 +13,73 @@
             if "source" in config:
-                self.source_part_name = config["source"]
+                source_part_name = config["source"]
             else:
-                self.source_part_name = config["name"]
-                if not "project" in config:
-                    raise Exception(
-                        "Enrich needs either the source part name or the source project name"
-                    )
+                source_part_name = config["name"]
+                if "project" not in config and "package" not in config:
+                    raise Exception("Enrich needs either the source part name or the source project name")
```

Ruff (0.8.2)

Create your own exception

(TRY002)